index.bs

```diff
@@ -316,56 +317,94 @@ Interface {#interface}
 =================
 
 <pre class="idl">
-dictionary NavigatorUABrandVersionDict {
-  required DOMString brand;
+dictionary NavigatorUABrandVersion {
+  DOMString brand;
   DOMString version;
 };
 
+dictionary UADataValues {
+  DOMString platform; 
+  DOMString platformVersion;
+  DOMString architecture;
+  DOMString model;
+};
+
 [Exposed=Window]
 interface NavigatorUAData {
-  readonly attribute FrozenArray&lt;NavigatorUABrandVersionDict&gt; brand;
+  readonly attribute FrozenArray&lt;NavigatorUABrandVersion&gt; uaList;
   readonly attribute boolean mobile;
-  Promise&lt;DOMString&gt; getPlatform();
-  Promise&lt;SOMString&gt; getPlatformVersion();
-  Promise&lt;DOMString&gt; getArchitecture();
-  Promise&lt;DOMString&gt; getModel();
+  Promise&lt;UADataValues&gt; getHighEntropyValues(sequence&lt;DOMString&gt; hints);
 };
 
 interface mixin NavigatorUA {
-  [SecureContext] NavigatorUAData getUserAgent();
+  [SecureContext] readonly attribute NavigatorUAData userAgentData;
 };
 Navigator includes NavigatorUA;
 
 </pre>
 
 Processing model {#processing}
 --------------
-<dfn method for="NavigatorUA"><code>getUserAgent()</code></dfn> method MUST run these steps:
+
+<h3 id="monkeypatch-html-windoworworkerglobalscope"><code>WindowOrWorkerGlobalScope</code></h3>
+
+Every {{WindowOrWorkerGlobalScope}} object has an associated <dfn for="WindowOrWorkerGlobalScope">UA list</dfn>, which is a <code><a interface>FrozenArray</a>&lt;<a dictionary>NavigatorUABrandVersion</a>></code>. It is initially the result of [=create a frozen array|creating a frozen array=] from the [=/list=] returned from running [=create UA list=].
```

Comment:
So, if you want "create UA list" to run only once per user agent, then you'll need some kind of novel architecture for that. As-written it will be re-run on each Window creation, with potentially different results.

Maybe "Each user agent has a UA list, which is created by running create UA list." Ideally you'd clarify that this is done once per ... user agent session?? ... until the browser gets updated??? ... but that doesn't seem strictly necessary, as just associating it with the UA will make it somewhat clear it should not get reset (e.g. reset on each WindowOrWorkerGlobalScope creation). 

Then, each WindowOrWorkerGlobalScope can have a UA list (or, UA list frozen array?), which is the result of running create a frozen array on the user agent's UA list.

Then, you can update other parts of the spec (e.g. the headers) to reference the user agent's UA list.

Reply:
done